integration_test/test/recording_endpoint_test.exs

```diff
@@ -179,4 +226,69 @@ defmodule Membrane.RTC.RecordingEndpointTest do
   defp validate_report(report_path) do
     assert report_path |> File.read!() |> byte_size() > 0
   end
+
+  defp setup_mock_http_request() do
+    pid = self()
+
+    expect(ExAws.Request.HttpMock, :request, 4, fn method, url, body, _headers, _opts ->
+      case %{method: method, url: url, body: body} do
+        %{
+          method: :post,
+          url: @url_prefix <> _rest,
```

Comment:
Why did you add a separate case instead of a separate clause of fn and here pattern match?